utils/common/src/main/java/brooklyn/util/javalang/Threads.java

```diff
@@ -44,7 +46,15 @@ public static boolean removeShutdownHook(Thread hook) {
             return Runtime.getRuntime().removeShutdownHook(hook);
         } catch (IllegalStateException e) {
             // probably shutdown in progress
-            log.debug("cannot remove shutdown hook "+hook+": "+e);
+            String text = Exceptions.collapseText(e);
+            if (text.contains("Shutdown in progress")) {
+                if (log.isTraceEnabled()) {
+                    log.trace("Could not remove shutdown hook "+hook+": "+text);
```

Comment:
Add trace and debug level, we should just include the entire stacktrace rather than truncating. Otherwise we risk losing important information for debugging problems that customers send via log files.